### ISSUE TITLE
Release handling

### ISSUE THREAD

**Issue by bslbckr** [ultical/ultical#31]
I think we should organize our releases a bit better, although we haven't had any trouble in this area, yet. What I could think of is the following:
- Branch `release` that collects planned features for the next release
- Branch `master` that is equivalent to the software running in 'production'
- Each time we merge from release into master:
  - we increment the versions inside the `pom.xml` and `package.json` to the next release-version
  - pull a tag from master named after the version to be released
  - change version to next development version (i.e. -SNAPSHOT for maven)
- Deployment should be done from the tag.

Of course, bug fixes should go into master directly, but preferrably using a branch. A bug-fix should only increment the bug-fix or patch-version part.

What do you think on this?

**Comment by bastrapp** [comment]
Generally I think it's a good idea to formalize and structure our development efforts. But with more or less 2 active developers it sounds like a lot of overhead.

It surely should be considered in a later phase of the project, when we plan releases based on issues and the development speed has slowed down a bit

**Comment by bslbckr** [comment]
Of course, we don't need planned releases, but at least we should know which software version was running to a current point in time. This could easily be achieved by incrementing the version number in the pom and tagging master.

**Comment by ChristianUlbrich** [comment]
GitFlow to the rescue. For releasing their should be some maven-release plugin. But, to keep things simple, I think simply adhering to GitFlow as a branching model is sufficient. I have used it on nearly every software project in the past and right now I am introducing it for a Client. You can easily adhere to it, by using either the git-flow commands or the built-in support in SourceTree.

https://github.com/petervanderdoes/gitflow-avh/wiki has everything you need, the original git-flow has the fundamental design flaw, that `git describe` does not work well (read: not at all) in develop branches.
